src/include/cell_inline.h

```diff
@@ -53,33 +53,62 @@ __cell_pack_value_validity(WT_SESSION_IMPL *session, uint8_t **pp, WT_TIME_WINDO
     ++*pp;
 
     flags = 0;
-    if (tw->start_ts != WT_TS_NONE) {
+    const bool pack_prepare_info_to_start = F_ISSET(S2BT(session), WT_BTREE_DISAGGREGATED) &&
+      tw->prepare && tw->prepare_ts != WT_TS_NONE &&
+      (!WT_TIME_WINDOW_HAS_STOP(tw) || tw->stop_txn == tw->start_txn);
+    const bool pack_prepare_info_to_stop = F_ISSET(S2BT(session), WT_BTREE_DISAGGREGATED) &&
+      tw->prepare && tw->prepare_ts != WT_TS_NONE && WT_TIME_WINDOW_HAS_STOP(tw);
+    wt_timestamp_t reference_ts = tw->start_ts;
+
```

Comment:
We can do something like:

WT_ASSERT(session, !pack_prepare_info_to_start || !pack_prepare_info_to_stop || tw->prepare_ts != WT_TS_NONE);